src/segment-loader.js

```diff
@@ -6,13 +6,11 @@ import {getMediaIndexForTime_ as getMediaIndexForTime, duration} from './playlis
 import videojs from 'video.js';
 import SourceUpdater from './source-updater';
 import {Decrypter} from './decrypter';
+import Config from './config';
```

Comment:
the same is true for any other global configuration though, I could see setting GOAL_BUFFER_LENGTH back to default on HLS dispose but on src change I think that global options should remain at what it was set to. We basically need HLS and SegmentLoader to share the value somehow so I went with this as it did not seem like we wanted to use crazy getters and setters.